region/tourian/main/Tourian Escape Room 3.json

```diff
@@ -505,38 +528,167 @@
-              "canBePatient"
+              "Plasma",
+              {"and": [
+                "h_hasBeamUpgrade",
+                "canBePatient"
+              ]},
+              "canBeVeryPatient"
+            ]}
+          ]},
+          {"and": [
+            {"or": [
+              "canBePatient",
+              {"enemyDamage": {
+                "enemy": "Tourian Space Pirate (all)",
+                "type": "contact",
+                "hits": 1
+              }}
+            ]},
+            {"or": [
+              "Morph",
+              "canTurnaroundAimCancel",
+              "canXRayTurnaround"
             ]}
           ]}
         ]}
       ],
+      "clearsObstacles": ["A"],      
+      "note": [
+        "Run from or Roll under the pirate lazers.",
+        "The Pirates will also not shoot if they come on screen while Samus is crouched."
+      ],
+      "devNote": [
+        "Technically it's not possible to kill the bottom Pirates while staying at node 2 (not dropping down);",
+        "but if we're not going down, then it doesn't matter that the bottom ones aren't killed."
+      ]
+    },
+    {
+      "link": [2, 2],
+      "name": "Kill Pirates Fast",
+      "requires": [
+        {"or": [
+          "Wave",
+          "ScrewAttack",
+          {"and": [
+            "Plasma",
+            "canXRayWaitForIFrames"
+          ]},
+          {"enemyKill": {
+            "enemies": [
+              [
+                "Tourian Space Pirate (all)",
+                "Tourian Space Pirate (all)"
+              ],
+              [
+                "Tourian Space Pirate (all)",
+                "Tourian Space Pirate (all)"
+              ],
+              ["Tourian Space Pirate (all)"]
+            ],
+            "explicitWeapons": ["Missile", "Super", "Ice+Plasma"]
+          }}
+        ]}
+      ],
+      "clearsObstacles": ["A"],
+      "note": "Kill the Pirates fast enough that they won't attack, or shoot them from above with Wave.",
+      "devNote": [
+        "Technically it's not possible to kill the bottom Pirates while staying at node 2 (not dropping down);",
+        "but if we're not going down, then it doesn't matter that the bottom ones aren't killed.",
+        "FIXME: The enemyKill requirements could be reduced if it is only needed to kill the top ones, which could be modeled as a separate obstacle."
+      ]
+    },
+    {
+      "link": [2, 2],
+      "name": "Leave Shinecharged",
+      "requires": [
+        {"obstaclesCleared": ["A"]},
+        "h_canShineChargeMaxRunway"
+      ],
       "exitCondition": {
         "leaveShinecharged": {
           "framesRemaining": 120
         }
       },
       "flashSuitChecked": true
     },
+    {
+      "link": [2, 2],
+      "name": "Leave With Mockball",
+      "requires": [
+        {"obstaclesCleared": ["A"]}
+      ],
+      "exitCondition": {
+        "leaveWithMockball": {
+          "remoteRunway": {
+            "length": 23,
```

Comment:
These full listed speed jumps feel like canTrickyJump.  But that's not different from many of these other remote runways that have to hit the door.